src/jwk-to-pem.js

```diff
@@ -3,6 +3,12 @@
 var ec = require('./ec'),
 	rsa = require('./rsa');
 
+/**
+ *
+ * @param {{kty:'EC', crv:string, d:string, x?:string, y?:string} | {kty:'RSA', e:string, n:string, d?:string, p?:string, q?:string, dp?:string, dq?:string, qi?:string}} jwk
```

Comment:
Don't think the EC one is quite right, but unfortunately you can't really define it properly. Could if private was it's own function I suppose.

```
opts.private === true: d, x?, y?
opts.private === false: x, y, d? OR d, x? y?
```

`{kty:'EC', crv:string, d:string } | {kty:'EC', crv:string, x:string, y:string}` is probably reasonable enough?

Reply:
OK, i defined it now in this way:

```js
// EC with d
{kty:'EC', crv:string, d:string, x?:string, y?:string}

// EC without d
{kty:'EC', crv:string, x:string, y:string}

// RSA
{kty:'RSA', e:string, n:string, d?:string, p?:string, q?:string, dp?:string, dq?:string, qi?:string}
```